### DISCUSSION TITLE
Slow simulation and OutOfMemoryError

### DISCUSSION THREAD

**Discussion by ItsMonish** [eclipse/mosaic#355] [Q&A]
Hey,just before anything I'm still learning MOSAIC. So I have tried to implement a custom scenario. So far I have followed up with the tutorials and documentation. Did the scenario folder structure, compiled a jar file and mapped the applications. When I run the scenario, the simulation is incrementing in order of nanoseconds and soon spits out a OutOfMemoryError. The simulation is set to cap at just 40 vehicles for 300s. 

Any idea what I should look into to resolve this? I may have overlooked something. Thanks for the help in advance.

**Comment by schwepmo** [comment]
Hey, are you running MOSAIC using the `mosaic.<sh|bat>`-script or do you use the integrated `MosaicStarter` out of your IDE?
If you're using the script you can adjust the script itself increasing the value of `javaMemorySizeXmx="2g"`.
If you're using the integrated starter your you can try to add `-Xmx10g` which allows for the process to allocate up to 10GB of ram.

Anyhow, out of the box MOSAIC isn't that memory intensive so maybe your application tries to allocate a lot of memory, which potentially points to a bug.

Hope that helps, if you have any follow-up questions feel free to ask!
